main_test.go

```diff
@@ -29,34 +32,57 @@ func TestStart(t *testing.T) {
 						So(server.dbDir, ShouldNotBeBlank)
 						So(server.port, ShouldNotBeEmpty)
 						So(server.cmd.Args[0], ShouldEndWith, "mongod")
-						So(server.cmd.Args[1], ShouldEqual, "--storageEngine")
-						So(server.cmd.Args[2], ShouldEqual, "ephemeralForTest")
+						So(server.cmd.Args[1], ShouldEqual, "--replSet")
+						So(server.cmd.Args[2], ShouldEqual, server.ReplicaSet())
 						So(server.cmd.Args[3], ShouldEqual, "--dbpath")
 						So(server.cmd.Args[4], ShouldEqual, server.dbDir)
 						So(server.cmd.Args[5], ShouldEqual, "--port")
-						So(server.cmd.Args[6], ShouldEqual, "0")
-					})
-					Convey("And the watcher process has run", func() {
-						expectedScript := fmt.Sprintf("while ps -o pid= -p %d; "+
-							"do sleep 1; "+
-							"done; "+
-							"kill -9 %d",
-							os.Getpid(), server.cmd.Process.Pid)
+						So(server.cmd.Args[6], ShouldEqual, strconv.Itoa(server.port))
 
-						So(server, ShouldNotBeNil)
-						So(server.watcherCmd, ShouldNotBeNil)
-						So(server.watcherCmd.Args[0], ShouldEndWith, "/bin/sh")
-						So(server.watcherCmd.Args[1], ShouldEqual, "-c")
-						So(server.watcherCmd.Args[2], ShouldEqual, expectedScript)
-					})
-					Convey("And the server accepts connections", func() {
-						client, err := mongo.Connect(testCtx, options.Client().ApplyURI(server.URI()))
-						So(err, ShouldBeNil)
-						So(client, ShouldNotBeNil)
-						So(client.Ping(testCtx, nil), ShouldBeNil)
+						Convey("And the watcher process has run", func() {
+							expectedScript := fmt.Sprintf("while ps -o pid= -p %d; "+
+								"do sleep 1; "+
+								"done; "+
+								"kill -9 %d",
+								os.Getpid(), server.cmd.Process.Pid)
+
+							So(server, ShouldNotBeNil)
+							So(server.watcherCmd, ShouldNotBeNil)
+							So(server.watcherCmd.Args[0], ShouldEndWith, "/bin/sh")
+							So(server.watcherCmd.Args[1], ShouldEqual, "-c")
+							So(server.watcherCmd.Args[2], ShouldEqual, expectedScript)
+
+							Convey("And the server accepts connections", func() {
+								client, err := mongo.Connect(testCtx, options.Client().ApplyURI(server.URI()))
+								So(err, ShouldBeNil)
+								So(client, ShouldNotBeNil)
+								So(client.Ping(testCtx, nil), ShouldBeNil)
+							})
+
+						})
 					})
 				})
 			})
 		})
 	}
 }
+
+func TestGetFreeMongoPort(t *testing.T) {
+	Convey("When getFreeMongoPort() is called n times, where n > 1", t, func() {
+		n := 10
+		defaultPort := "27017"
+
+		Convey("A free, non default and usable port should be returned every time", func() {
+			for i := 0; i < n; i++ {
+				port := getFreeMongoPort()
+				So(port, ShouldNotEqual, defaultPort)
+
+				l, e := net.Listen("tcp", "localhost:"+port)
+				So(e, ShouldBeNil)
+				So(l, ShouldNotBeNil)
+				e = l.Close()
+				So(e, ShouldBeNil)
```

Comment:
Is this part of the test? Not sure we need to assert if it closes? Maybe just add a defer function?